compiler/src/iree/compiler/Dialect/LinalgExt/IR/TilingInterfaceImpl.cpp

```diff
@@ -388,6 +388,62 @@ GatherOp::generateResultTileValue(OpBuilder &builder, unsigned resultNumber,
   return getTiledImplementation(builder, offsets, sizes);
 }
 
+LogicalResult GatherOp::generateScalarImplementation(OpBuilder &b, Location loc,
+                                                     ValueRange ivs) {
+  auto indexDepth = getIndexDepth();
+  Value result = b.create<memref::LoadOp>(loc, getOutput(), ivs);
+  SmallVector<Value> starts;
+  SmallVector<Value> loadIndices;
+  append_range(loadIndices, ivs.take_front(getBatchRank()));
```

Comment:
nit: I think this should also work
```suggestion
  SmallVector<Value> loadIndices(ivs.take_front(getBatchRank()));
```

Reply:
I changed the merge base too early on this PR. These changes are from https://github.com/iree-org/iree/pull/20464 which got blocked by flaky CI. I added a commit that addresses them there.